jetty-websocket/jetty-websocket-tests/src/test/java/org/eclipse/jetty/websocket/tests/client/ClientConnectTest.java

```diff
@@ -456,4 +456,39 @@ public void testConnectionTimeout_Concurrent() throws Exception
             }
         }
     }
+
+    @Test
+    public void testIdleTimeout() throws Exception
+    {
+        client.getHttpClient().setIdleTimeout(-1);
+        client.setMaxIdleTimeout(Duration.ofSeconds(3).toMillis());
+
+        CloseTrackingEndpoint cliSock = new CloseTrackingEndpoint();
+        try (ServerSocket serverSocket = new ServerSocket())
+        {
+            InetAddress addr = InetAddress.getByName("localhost");
+            InetSocketAddress endpoint = new InetSocketAddress(addr, 0);
+            serverSocket.bind(endpoint, 1);
+            int port = serverSocket.getLocalPort();
+            URI wsUri = URI.create(String.format("ws://%s:%d/", addr.getHostAddress(), port));
+            Future<Session> future = client.connect(cliSock, wsUri);
+
+            // Accept the connection, but do nothing on it (no response, no upgrade, etc)
+            serverSocket.accept();
+
+            // The attempt to get upgrade response future should throw error
+            Exception e = assertThrows(Exception.class,
+                () -> future.get(5, TimeUnit.SECONDS));
```

Comment:
If the 5 seconds wait here is related to the idle timeout, then it should be `2 * idleTimeout`.